### PR TITLE
fix (ras-acc): prevent NYP, coupon submission from clearing credit card fields

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR moves setting and updating the 'Complete Transaction' button price from PHP to JS. This is to prevent an issue where submitting a coupon or NYP after filling in the credit card fields would cause the credit card fields to be cleared.

See: 1207817176293825-as-1208255803261156.

### How to test the changes in this Pull Request:

1. Start on epic/ras-acc.
2. Enable coupons, and set up at least one regular coupon.
3. Set up Checkout button blocks for a simple product and a NYP product, plus a donate block.
4. Set up Stripe Gateway and WooPayments as your payment options.
5. As a reader, run through a transaction using the simple product.
6. On the second screen, fill out your payment information first. 
7. Fill in your coupon, and Apply.
8. Note that the credit card fields reset.
9. Repeat steps 5-8 with the NYP product, and with both payment gateways.
10. Apply this PR and run `npm run build`.
11. Repeat steps 5-8 with both your simple and NYP products, and both payment processors. When you do this, confirm that the credit card values remain in place, and also confirm that the Confirm Transaction button at the bottom updates with the correct price. 
12. Try applying your coupon, then removing it, and checking the price in the Confirm Transaction button each time. 
13. Run a transaction with the Donate Button block, and confirm the button on the second screen just says "Donate Now". 
14. Smoke test a few other product options, like a subscription product or variable product, and make sure the Confirm Transaction button displays the correct initial payment amount.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/modal-checkout/index.js [hunk 1]
```diff
@@ -128,6 +128,48 @@ domReady(
 					$( '.payment_methods' ).after( $el );
 				} );
 
+				/**
+				 * Get updated cart total to update the "Place Order" button.
+				 */
+				function returnUpdatedCartTotal() {
```
Comment (chickenn00dle) [inline src/modal-checkout/index.js]: non-blocking nit: might be clearer to name this something like `getUpdatedCartTotal`

**Review by chickenn00dle — APPROVED** [review]
Nice work @laurelfulford! Looks good for the most part. Left a small non-blocking nit.

I also ran into a very edge case issue where when a product has both coupons AND nyp enabled, submitting a new NYP value will override the coupon entirely. However, this seems to be happening on `epic/ras-acc` as well so will approve and [create a new task for this](https://app.asana.com/0/1207817176293825/1208344897866336/f).

**Comment by laurelfulford** [comment]
> I also ran into a very edge case issue where when a product has both coupons AND nyp enabled, submitting a new NYP value will override the coupon entirely.

Eeeeep good catch! 😱 I ran _so_ many checkouts testing this and still missed it -- so glad you caught it!

File: src/modal-checkout/index.js (hunk 1 above)
Comment (laurelfulford) [reply src/modal-checkout/index.js]: Good call! This has been updated in [013161e](https://github.com/Automattic/newspack-blocks/pull/1881/commits/013161edda5b51a1aa8d32e4f0b5daafd5b50142).

**Pull request merged**